src/test/java/io/pravega/connectors/flink/FlinkPravegaTableFactoryTest.java

```diff
@@ -0,0 +1,272 @@
+                .withFormat(JSON)
+                .withSchema(SCHEMA)
+                .inAppendMode();
+
+        final Map<String, String> propertiesMap = DescriptorProperties.toJavaMap(testDesc);
+
+        FlinkPravegaTableFactoryBase tableFactoryBase = new FlinkPravegaStreamTableSourceFactory();
+        tableFactoryBase.createFlinkPravegaTableSource(propertiesMap);
+    }
+
+    /**
+     * Stream table source expects 'update-mode' configuration to be passed.
+     */
+    @Test (expected = NoMatchingTableFactoryException.class)
+    public void testMissingStreamMode() {
+
+        Pravega pravega = new Pravega();
+        Stream stream = Stream.of(SCOPE, STREAM);
+
+        pravega.tableSourceReaderBuilder()
+                .forStream(stream)
+                .withPravegaConfig(PRAVEGA_CONFIG);
+
+        final TestTableDescriptor testDesc = new TestTableDescriptor(pravega)
+                .withFormat(JSON)
+                .withSchema(SCHEMA);
+
+        final Map<String, String> propertiesMap = DescriptorProperties.toJavaMap(testDesc);
+
+        final TableSource<?> source = TableFactoryService.find(StreamTableSourceFactory.class, propertiesMap)
+                .createStreamTableSource(propertiesMap);
+        TableSourceUtil.validateTableSource(source);
+    }
+
+    /**
+     * For sink, stream name information is mandatory.
+     */
+    @Test (expected = IllegalStateException.class)
+    public void testMissingStreamNameForWriter() {
+        Pravega pravega = new Pravega();
+
+        pravega.tableSinkWriterBuilder()
+                .withRoutingKeyField("name");
+
+        final TestTableDescriptor testDesc = new TestTableDescriptor(pravega)
+                .withFormat(JSON)
+                .withSchema(SCHEMA)
+                .inAppendMode();
+
+        final Map<String, String> propertiesMap = DescriptorProperties.toJavaMap(testDesc);
+        final TableSink<?> sink = TableFactoryService.find(StreamTableSinkFactory.class, propertiesMap)
+                .createStreamTableSink(propertiesMap);
+        assertNotNull(sink);
+    }
+
+    /**
+     * For sink, routingkey-field-name information is mandatory.
+     */
+    @Test (expected = ValidationException.class)
+    public void testMissingRoutingKeyForWriter() {
+        Pravega pravega = new Pravega();
+        Stream stream = Stream.of(SCOPE, STREAM);
+
+        pravega.tableSinkWriterBuilder()
+                .forStream(stream)
+                .withPravegaConfig(PRAVEGA_CONFIG);
+
+        final TestTableDescriptor testDesc = new TestTableDescriptor(pravega)
+                .withFormat(JSON)
+                .withSchema(SCHEMA)
+                .inAppendMode();
+
+        final Map<String, String> propertiesMap = DescriptorProperties.toJavaMap(testDesc);
+        final TableSink<?> sink = TableFactoryService.find(StreamTableSinkFactory.class, propertiesMap)
+                .createStreamTableSink(propertiesMap);
+        assertNotNull(sink);
```

Comment:
If we get here, then it means that the test failed because we expect it to throw, right? Shouldn't we directly `fail()` here? The same comment is valid for pretty much every other test case in this class.

Reply:
fixed.